graph-kiota/kiota.md

```diff
@@ -0,0 +1,180 @@
+            sendNoResponseContentAsync(requestInfo: RequestInformation, responseHandler: ResponseHandler | undefined): Promise<void>;
+            enableBackingStore(backingStoreFactory?: BackingStoreFactory | undefined): void;
+        }
+        ````
+        - Serialization interfaces
+        - Authentication: `IAuthProvider`, abstract classes- `anonymousAuthenticationProvider` and `BearerTokenAuthneticationProvider`
+        - Utils
+        - ResponseHandler interface
+        - MiddlewareOption class
+        - RequestInformation
+        - HttpMethod Enum
+
+- Kiota core
+    - This is generic core client library (functionality or purpose similar to Microsoft Graph core libraries)
+    [Kiota core dotnet](https://github.com/microsoft/kiota/tree/main/http/dotnet/httpclient)
+    [Kiota core java](https://github.com/microsoft/kiota/tree/main/http/java/okhttp)
+    [Kiota core typescript](https://github.com/microsoft/kiota/tree/main/http/typescript/fetch)
+    - This library contains the following:
+        -  HttpCore class implementing the abstraction HttpCore interface.
+        -  HttpClientbuilder (.NET, Java) or HTTPClient class (TS) (processes and records the middleware chain used while making requests.)
+        -  Middlewares: AuthHandler, Retry, Redirect, Compression, Caching, Telemetry
+    
+```
+class HttpCore implements kiota.abstractions.HttpCore {
+    constructor(authProvider, parseNodeFactory, serializationWriteFactory, httpClient);
+
+    sendAsync<ModelType extends Parsable>(requestInfo: RequestInformation, type: new() => ModelType, responseHandler: ResponseHandler | undefined): Promise<ModelType>;
+    sendCollectionAsync<ModelType extends Parsable>(requestInfo: RequestInformation, type: new() => ModelType, responseHandler: ResponseHandler | undefined): Promise<ModelType[]>;
+    sendPrimitiveAsync<ResponseType>(requestInfo: RequestInformation, responseType: "string" | "number" | "boolean" | "Date" | "ReadableStream", responseHandler: ResponseHandler | undefined): Promise<ResponseType>;
+    sendNoResponseContentAsync(requestInfo: RequestInformation, responseHandler: ResponseHandler | undefined): Promise<void>;
+    enableBackingStore(backingStoreFactory?: BackingStoreFactory | undefined): void;
+}
+
+class HttpClient{
+    middlewares : Middlewares();
+    send() //calls the MiddlewareChain to make a  request with the native HttpClient;
+}
+```
+
+Kiota Generated API Client (This will be the GraphServiceClient)
+
+```
+class APIClient {
+
+    httpCoreInstance : HttpCore
+
+    SampleRequestBuilder(httpCoreInstance);
+    ExampleRequestBuilder(httpCoreInstance);
+}
+```
+
+- Kiota serializations 
+    - This library contains implementation of the kiota abstractions serialization interfaces
+    - Currently targets `JSON` format.
+    - Other formats to consider:
+        - Multipart payload
+        - Base64 encoded string
```

Comment:
the case where it's part of a json payload should be handled today already. If the payload is only base64 encoded, this needs to be reviewed.